pkg/operation/botanist/controlplane.go

```diff
@@ -321,12 +346,13 @@ func (b *Botanist) ScaleGardenerResourceManagerToOne(ctx context.Context) error
 	return kubernetes.ScaleDeployment(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameGardenerResourceManager), 1)
 }
 
-// PrepareControlPlaneDeploymentsForMigration scales the kube-apiserver, kube-controller-manager and gardener-resource-manager to zero and deletes the hvpa for the kube-apiserver
-func (b *Botanist) PrepareControlPlaneDeploymentsForMigration(ctx context.Context) error {
+// PrepareKubeAPIServerForMigration deletes the kube-apiserverand deletes its hvpa
+func (b *Botanist) PrepareKubeAPIServerForMigration(ctx context.Context) error {
```

Comment:
Maybe it will be better to put the deletion of the hvpa in separated function as the deletion of the kupe-apiserver deployment

Reply:
I am not sure what will be the actual benefit except that it could be reused in future. But I think in most cases deleting the KubeAPI server will require cleaning the hvpa as well, so you will need to add both steps again? WDYT?